extensions/kafka/src/main/java/io/deephaven/kafka/KafkaTools.java

```diff
@@ -2189,8 +2299,8 @@ public long consume(List<? extends ConsumerRecord<?, ?>> consumerRecords) {
         }
 
         @Override
-        public void acceptFailure(@NotNull Throwable cause) {
-            streamToBlinkTableAdapter.acceptFailure(cause);
+        public void acceptFailure(@NotNull final Throwable cause) {
+            adapter.propagateFailure(cause);
```

Comment:
Love it - makes this much cleaner. Really, `StreamPublisher` is an interface that is _meant_ to be extended with proper "here is new data" / "here is an error". We _might_ consider adding and documenting `propogateFailure` on `StreamPublisher` itself if it makes sense for all impls.

(In a similar way, see other comment about if shutdown() should "propogate" to consumer.)

Reply:
Generally, the `StreamPublisher` interface is an API for the `StreamConsumer` to interact with. I don't want to add propagation to that, it's an implementation detail of the publisher.